### PR TITLE
Update Main.cs

### PR DESCRIPTION
**Pull request by Jekenius**
Added work with exceptions. Added question for LINQ made by Alexey.

### CODE REVIEW

File: FileWeightChecker/Main.cs [hunk 1]
```diff
@@ -8,17 +8,17 @@
 
 public class FileLength
 {
-    public static void DirectoryMassCheck(DirectoryInfo CurrentDirectory, DirectoryInfo StartingDirectory, string OutputFileName)
```
Comment (baradgur) [inline FileWeightChecker/Main.cs:11]: За одно только переименование методов - жирный плюс!

File: FileWeightChecker/Main.cs [hunk 2]
```diff
@@ -38,8 +38,29 @@ public static void AverageAndMaxCheck(string InputFileName, string OutputFileNam
         double Average = 0.0;
 
         string[] AllLines = File.ReadAllLines (InputFileName);
-        foreach(string line in AllLines)
+
+        /*
+        //пример нахождения максимального значения с помощью LINQ
+        
+        //What are these question marks doing? They should be some check for null filled lines here but what is their exact purpose?
```
Comment (baradgur) [inline FileWeightChecker/Main.cs]: верно, это conditional access operator
https://stackoverflow.com/questions/25666993/c-sharp-null-propagating-operator-conditional-access-expression-if-blocks

ты прав, не все из этих проверок могут сработать, например
.Trim(new char[] {' '})) не может null вернуть никак, только пустую строку
по факту это пример паранойи, я просто по быстрому писал, привычка

File: FileWeightChecker/Main.cs [hunk 3]
```diff
@@ -48,38 +69,69 @@ public static void AverageAndMaxCheck(string InputFileName, string OutputFileNam
...
+                sw.WriteLine($"Avegage length equals {Average} bytes");
                 sw.Close(); 
             }
 
     }
 
     public static void Main()
     {
+        //обработку ошибок бы хотелось
+        //Done. At least places where i know they may happen.
```
Comment (baradgur) [inline FileWeightChecker/Main.cs]: it's a normal practice to use one big try catch in console programs. At least here

File: FileWeightChecker/Main.cs [hunk 4]
```diff
@@ -48,38 +69,69 @@ public static void AverageAndMaxCheck(string InputFileName, string OutputFileNam
...
+        {
+            OutputFileDirectory = Console.ReadLine();
+            Directory.CreateDirectory(OutputFileDirectory);
+        }
+        catch
+        {
+            //If trying again is not the best option use this:
+            //return;
+            Console.WriteLine("Incorrect directory name. Try again.");
+            goto TryAgainOutputDirectory;
```
Comment (baradgur) [inline FileWeightChecker/Main.cs]: please use while here, never use goto. I saw only 1 good usage of goto in my life

File: FileWeightChecker/Main.cs [hunk 5]
```diff
@@ -48,38 +69,69 @@ public static void AverageAndMaxCheck(string InputFileName, string OutputFileNam
...
+        try
+        {
+            OutputFileDirectory = Console.ReadLine();
+            Directory.CreateDirectory(OutputFileDirectory);
+        }
+        catch
+        {
+            //If trying again is not the best option use this:
+            //return;
+            Console.WriteLine("Incorrect directory name. Try again.");
```
Comment (baradgur) [inline FileWeightChecker/Main.cs]: what if user does not have rights to access path? same message?

Comment (baradgur) [reply FileWeightChecker/Main.cs]: and after 10 tries user wants to exit but can't

File: FileWeightChecker/Main.cs [hunk 6]
```diff
@@ -48,38 +69,69 @@ public static void AverageAndMaxCheck(string InputFileName, string OutputFileNam
...
+            goto TryAgainStrartingDirectory;        
+        }   
         
-        DirectoryMassCheck(StartingDirectory, StartingDirectory, AllMassesOutputFile.Name);
-        AverageAndMaxCheck(AllMassesOutputFile.Name, AverageAndMaxMassOutputFile.Name);
-
+        DirectoryLengthCheck(StartingDirectory, StartingDirectory, AllLengthsOutputFile.Name);
```
Comment (baradgur) [inline FileWeightChecker/Main.cs]: can exception happen here/ which ones?

File: FileWeightChecker/Main.cs (hunk 5 above)
Comment (Jekenius) [reply FileWeightChecker/Main.cs]: Got it. Will be re-done,

File: FileWeightChecker/Main.cs (hunk 4 above)
Comment (Jekenius) [reply FileWeightChecker/Main.cs]: Like to use try-catch inside while until there is no exceptions?

Comment (baradgur) [reply FileWeightChecker/Main.cs]: yes, but set the max number of tries

**Comment by Jekenius** [comment]
Added new version of exception handling trying to apply all advices given.

**Comment by baradgur** [comment]
d

**Review by baradgur — APPROVED** [review]
ok, lgtm

**Review by baradgur — APPROVED** [review]

**Pull request merged**